docs/workflow-executors.md

```diff
@@ -79,3 +79,47 @@ The executor to be used in your workflows can be changed in [the configmap](./wo
 * [Doesn't work for Windows containers](https://kubernetes.io/docs/setup/production-environment/windows/intro-windows-in-kubernetes/#v1-pod).
 
 [https://kubernetes.io/docs/tasks/configure-pod-container/share-process-namespace/](https://kubernetes.io/docs/tasks/configure-pod-container/share-process-namespace/)
+
+## Emissary (emissary)
+
+![alpha](assets/alpha.svg)
+
+> v3.1 and after
+
+This is the most fully featured executor.
+
+This executor works very differently to the others. It mounts an empty-dir on all containers at `/var/run/argo`. The main container command is replaces by a new binary `emissary` which starts the original command in a sub-process and when it is finished, captures the outputs:
+
+The init container creates these files:
+
+* `/var/run/argo/argoexec` The binary, copied from the `argoexec` image.
+* `/var/run/argo/template` A JSON encoding of the template.
+
+In the main container, the emissary creates these files: 
+
+* `/var/run/argo/outputs/parameters/${path}` All output parameters are copied here, e.g. `/tmp/message` is moved to /var/run/argo/outputs/parameters/tmp/message`.  
+* `/var/run/argo/outputs/artifacts/${path}.tgz` All output artifacts are copied here, e.g. `/tmp/message` is moved to /var/run/argo/outputs/artifacts/tmp/message.tgz`.  
+* `/var/run/argo/ctr/${containerName}/stderr` A copy of stderr. 
+* `/var/run/argo/ctr/${containerName}/stdout`  A copy of stdout.
+
+The wait container write the exit code:
+
+* `/var/run/argo/ctr/${containerName}/exitcode` The container exit code.
+
+The wait container can create one file itself, used for terminating the sub-process:
+
+* `/var/run/argo/ctr/${containerName}/signal` The emissary binary listens to changes in this file, and signals the sub-process with the signal found in this file.
+
+* Reliability:
+  * Least well-tested.
+  * Least popular.
+* More secure:
+  * No `privileged` access
+  * Cannot escape the privileges of the pod's service account
+  * Can [`runAsNonRoot`](workflow-pod-security-context.md).
+* Scalable:
+  * It reads and writes to and from the container's disk and does not use any network APIs.
```

Comment:
With one exception here where the template is resource type, which is something we use heavily and still working on customizing/optimizing, e.g. see https://github.com/argoproj/argo/issues/4467#issuecomment-775191432.